### ISSUE TITLE
Enable TaskTable to use API requests to filter tasks

### ISSUE THREAD

**Issue by lowellrex** [department-of-veterans-affairs/caseflow#11309]
Spun out of #11054.

The `TaskTable` component currently manages filtering only on the front-end. This ticket exists to enable us to pass filters from the front-end as part of the pagination API requests we make so that we can filter large queues. At some point we will migrate all queues to using the pagination API but are not doing that migration in a single step so we need to support both methods at the same time.

## 🚫 Blocked by
* [x] Allow organization queue table configuration to accept filtering (#11307)
* [x] Include filterable values in queue config response to front-end (#11509)
* [x] Enable remaining three task queue filters on back-end (#11888)

## Acceptance criteria
* [ ] `TaskTable` component's pagination, filtering, and sorting use API requests instead of happening on the front-end.
* [x] Use options returned by the back-end (#11509) to build the filter dropdown menus

## Non-goals
* In order to reduce the complexity of this ticket, let's not save any tasks in state initially. Caching will happen as part of #11310

**Comment by hschallhorn** [comment]
Added #11509 as a blocker as the back end will provide the values of what we will filter on

**Comment by lowellrex** [comment]
Noting here that the case type filter filters on two different columns:
1. `cached_appeal_attributes.case_type` for all case types except AOD
2. `cached_appeal_attributes.is_aod` for AOD

This complicates matters slightly when unpacking the filter on the back-end after it has been sent from the front-end.

**Comment by lowellrex** [comment]
@hschallhorn heads up that I added an additional acceptance criteria
> * [ ] Use options returned by the back-end (#11509) to build the filter dropdown menus

and another ticket that blocks this one
> * [ ] Enable remaining three task queue filters on back-end (#11888)
